tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/swtbot/SWTBotMenuHelper.java

```diff
@@ -64,4 +81,17 @@ public SWTBotMenu findMenu(final SWTBotMenu parentMenu,
             return new SWTBotMenu(menuItem);
         }
     }
+
+    public SWTBotMenu findWorkbenchMenu(final SWTWorkbenchBot bot, final String menuString) {
+        for (SWTBotShell shell : bot.shells()) {
```

Comment:
If I understand correctly you want to make sure the Eclipse window is the one that's active right?
If so, SWTBot FAQ shows a way to do that, which is easier than this, and does not require to distinguish between Eclipse and Maven build (https://wiki.eclipse.org/SWTBot/Troubleshooting) something like this should force the focus on the Eclipse workbench:

```java
UIThreadRunnable.syncExec(new VoidResult() {
  public void run() {
    PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell().forceActive();
  }
});
```

Reply:
I don't want to use the focus as much as possible, because this can be a root of problems, as seen in #152.

Reply:
OK, but if you need to access the workbench shell at least do that with the Eclipse API and not based on the name of the window (which is different in Eclipse and in Maven). Something similar to what I showed above. For example, this is taken from `org.eclipse.swtbot.eclipse.finder.DefaultWorkbench.getActiveWorkbenchWindowShell()` (Once you have a `Shell` you can construct an `SWTBotShell`):

```java
	private IWorkbenchWindow getActiveWorkbenchWindow() {
		return UIThreadRunnable.syncExec(bot.getDisplay(), new Result<IWorkbenchWindow>() {
			@Override
			public IWorkbenchWindow run() {
				return PlatformUI.getWorkbench().getActiveWorkbenchWindow();
			}
		});
	}

	private Widget getActiveWorkbenchWindowShell() {
		return getActiveWorkbenchWindow().getShell();
	}
```

you can see that the workbench window can be accessed programmatically independently from the window text.

Reply:
Please, also have a look at https://github.com/pitest/pitclipse/pull/152#issuecomment-896862926
from my experience when I get the test failure finding the menu is because SWTBot does not find the shell active (typically the Eclipse workbench) that's why I'm forcing the activation of the workbench.